### PR TITLE
feat(browser): add PDF support using pypdf

### LINKED ISSUE #242: PDF support
I've been working on reading them with the browser tool here: https://github.com/ErikBjare/gptme/pull/196

Anthropic just announced PDF support: https://x.com/alexalbert__/status/1852394000101323193

@simonw added it to llm here: https://x.com/simonw/status/1852423593084498409

The implementation I had in mind is model-agnostic, but doesn't include the vision component (probably good enough in 80% of cases), but Anthropics solution seems to be fully multimodal.

### PR DESCRIPTION
**Pull request by coop-must-build**
## Summary

Implements PDF reading support in the browser tool, addressing Issue #242.

## Changes

- Added `pypdf` dependency to `pyproject.toml`
- Implemented PDF detection by URL extension and Content-Type header
- Added `_read_pdf_url()` function for text extraction using pypdf
- Updated `read_url()` to automatically handle PDFs
- Added tests for PDF reading (arxiv) and PDF URL detection
- Updated browser tool documentation with PDF examples

## Technical Approach

- Pure Python solution using pypdf - no external system dependencies
- Automatic PDF detection by `.pdf` extension or `application/pdf` Content-Type
- Page-by-page text extraction with page markers
- Error handling for empty/invalid PDFs

## Testing

- Test: Read arxiv PDF and extract text
- Test: PDF URL detection by extension

Closes #242

----

> [!IMPORTANT]
> Adds PDF reading support to the browser tool using `pypdf`, with detection by URL extension or Content-Type, and updates documentation and tests.
> 
>   - **Behavior**:
>     - Adds PDF reading support to `read_url()` in `browser.py` using `pypdf`.
>     - Detects PDFs by `.pdf` extension or `application/pdf` Content-Type.
>     - Extracts text page-by-page with markers; handles empty/invalid PDFs.
>   - **Dependencies**:
>     - Adds `pypdf` to `pyproject.toml` for PDF text extraction.
>   - **Testing**:
>     - Adds `test_read_url_arxiv_pdf()` and `test_pdf_url_detection()` in `test_browser.py`.
>   - **Documentation**:
>     - Updates `browser.md` with PDF reading examples and notes.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=gptme%2Fgptme&utm_source=github&utm_medium=referral)<sup> for cc254d9a99d670df3b32c7e06a2d688c8ef5ed08. You can [customize](https://app.ellipsis.dev/gptme/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -43,6 +43,7 @@ tomlkit = "*"
 typing-extensions = "*"
 platformdirs = "^4.3"
 lxml = "*"
+pypdf = "^5.1.0"  # For PDF text extraction in browser tool
```
[the remark this answers is not in the document]
Comment (coop-must-build) [reply pyproject.toml:46]: ✅ Fixed in commit cf7de0c84 - added requests as explicit dependency with version ^2.32

**Review by TimeToBuildBob — APPROVED** [review]
Nice work @coop-must-build! 🎉

This is a solid implementation that expands gptme's capabilities for working with research papers and documents. Clean separation between PDF detection and reading, good error handling for edge cases (missing pypdf, empty/image-only PDFs), and comprehensive tests.

A few things I particularly liked:
- **Graceful degradation**: Clear error message when pypdf isn't installed
- **Content-Type fallback**: Smart detection via HEAD request for URLs that don't end in .pdf
- **Page markers**: Makes it easy to navigate extracted text

Welcome to the gptme community! Looking forward to your future contributions.
